tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -370,3 +409,106 @@ def test_pod_restart_node_nw_down(self):
+        LOGGER.info("Step 3: Performed READs/Verify on data written in healthy cluster.")
+        LOGGER.info("Step 4: Create new bucket and perform WRITEs/READs/Verify with variable "
+                    "object sizes in degraded mode")
+        if CMN_CFG["dtm0_disabled"]:
+            t_t = str(perf_counter_ns())
+            self.test_prefix_deg = f'test-45510-deg-{t_t}'
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipcleanup=True, setup_s3bench=False)
+        else:
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix,
+                                                        skipcleanup=True, setup_s3bench=False)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Performed WRITEs/READs/Verify with variable sizes objects in "
+                    "degraded mode")
+        LOGGER.info("Step 5: Restore data and server pod and check cluster status.")
+        for pod in self.pod_dict:
+            self.restore_method = self.pod_dict.get(pod)[-1]
+            resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                           restore_method=self.restore_method,
+                                           restore_params={
+                                               "deployment_name": self.pod_dict.get(pod)[-2],
+                                               "deployment_backup": self.deployment_backup,
+                                               "num_replica": self.pod_dict.get(pod)[2],
+                                               "set_name": self.pod_dict.get(pod)[1]})
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        LOGGER.info("Step 5: Successfully started data and server pod and cluster is online.")
+        self.restore_pod = False
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 6: Perform READs and verify DI on the data written in degraded mode")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 6: Successfully run READ/Verify on data written in degraded mode")
+        LOGGER.info("Step 7: Perform READ/Verify on data written in healthy mode")
```

Comment:
data written with buckets created in healthy mode, as it might also have data written in degraded mode